share/ruby-build/jruby-dev

```diff
@@ -0,0 +1,13 @@
+case $(uname -s) in
+Linux)
+  install_package "jruby-head" "https://github.com/ruby/jruby-dev-builder/releases/latest/download/jruby-head-ubuntu-20.04.tar.gz" jruby
+  ;;
+Darwin)
+  use_homebrew_openssl
+  install_package "jruby-head" "https://github.com/ruby/jruby-dev-builder/releases/latest/download/jruby-head-macos-latest.tar.gz" jruby
```

Comment:
The archive is the same on Linux and macOS, but there might be differences in the future so this sounds good.
Those URLs are not guaranteed to be stable forever, but they will probably be relatively stable for ruby/setup-ruby, and we can always update them.

Reply:
Self-note, discussion about platform-specific archives or not: https://github.com/jruby/jruby/issues/6092#issuecomment-962623179